src/balance.cpp

```diff
@@ -152,16 +150,15 @@ void Balance::command(int narg, char **arg)
         user_xsplit[procgrid[0]] = 1.0;
       }
     } else if (strcmp(arg[iarg],"y") == 0) {
-      if (style != -1 && style != XYZ)
-        error->all(FLERR,"Illegal balance command");
+      if ((style != -1) && (style != XYZ))
+        error->all(FLERR, iarg, "Must not combine multiple balancing styles");
       style = XYZ;
+      if (iarg+2 > narg) utils::missing_cmd_args(FLERR,"balance y", error);
       if (strcmp(arg[iarg+1],"uniform") == 0) {
-        if (iarg+2 > narg) error->all(FLERR,"Illegal balance command");
         yflag = UNIFORM;
         iarg += 2;
       } else {
-        if (1 + procgrid[1]-1 > narg)
-          error->all(FLERR,"Illegal balance command");
+        if (1 + procgrid[1]-1 > narg) utils::missing_cmd_args(FLERR,"balance y", error);
```

Comment:
Shouldn't these checks add `iarg`? If you have a 4 x 4 x 1 grid with the command:
`balance 0.9 x 0.1 0.1 0.1 y 0.1`
I think this check would miss the error.

Reply:
yup. well spotted. this appears to be a rather old bug that has survived a long time...